src/components/DocumentForm.tsx

```diff
@@ -1,19 +1,19 @@
-import { useContext } from "react";
 import { wrapDocument } from "@govtechsg/open-attestation";
-import { AppContext } from "../contexts/AppContext";
-import { AccountContext } from "../contexts/AccountContext";
+import { useAccountContext } from "../contexts/AccountContext";
+import { useDocumentStoreContext } from "../contexts/DocumentStoreContext";
+import { useDnsContext } from "../contexts/DnsContext";
+import { useStatusContext } from "../contexts/StatusContext";
+import { useStepContext } from "../contexts/StepContext";
+import { useWrappedDocumentContext } from "../contexts/WrappedDocumentContext";
 import { issueDocument } from "../services/document-store";
 
 export const DocumentForm = () => {
-  const {
-    dns,
-    documentStoreAddress,
-    status,
-    setStatus,
-    setCurrentStep,
-    setWrappedDocument,
-  } = useContext(AppContext);
-  const { signer } = useContext(AccountContext);
+  const { documentStoreAddress } = useDocumentStoreContext();
+  const { dns } = useDnsContext();
+  const { status, setStatus } = useStatusContext();
+  const { setCurrentStep } = useStepContext();
+  const { setWrappedDocument } = useWrappedDocumentContext();
+  const { signer } = useAccountContext();
```

Comment:
yah so this is where people will argue alot about.
whether the previous code has better intention than this 1 here.

for my way of thinking i find that yes the improved version is more verbose, but everything is segregated in their own contexts and if something goes wrong in any of the contexts, if we had like a monitoring tool like sentry it should leave an  error trace there.

whereas for the left 1, 1 will need to dive abit deeper and find out which of the combined values are at fault.